src/components/connection_handler/include/connection_handler/connection_handler_impl.h

```diff
@@ -182,13 +183,33 @@ class ConnectionHandlerImpl
    * \param hash_id pointer for session hash identifier
    * \return uint32_t Id (number) of new session if successful, otherwise 0.
    */
+  // DEPRECATED
   virtual uint32_t OnSessionStartedCallback(
       const transport_manager::ConnectionUID connection_handle,
       const uint8_t session_id,
       const protocol_handler::ServiceType& service_type,
       const bool is_protected,
       uint32_t* hash_id);
 
+  /**
+   * \brief Callback function used by ProtocolHandler
+   * when Mobile Application initiates start of new session.
+   * Result must be notified through NotifySessionStartedResult().
+   * \param connection_handle Connection identifier within which session
+   * has to be started.
+   * \param sessionId Identifier of the session to be start
+   * \param service_type Type of service
+   * \param protocol_version Version of protocol
+   * \param is_protected would be service protected
+   * \param params configuration parameters specified by mobile
+   */
+  virtual void OnSessionStartedCallback(
+      const transport_manager::ConnectionUID connection_handle,
+      const uint8_t session_id,
+      const protocol_handler::ServiceType& service_type,
+      const bool is_protected,
+      const BsonObject* params);
+
   // DEPRECATED
   uint32_t OnSessionEndedCallback(
       const transport_manager::ConnectionUID connection_handle,
```

Comment:
This is the version of the function that is in master, the definition above the one that you added was added in develop